src/FixedPointNumbers.jl

```diff
@@ -42,25 +40,198 @@ export
     # Functions
     scaledual
 
+# T => BaseType
+# f => Number of Bytes reserved for fractional part
+immutable FixedPoint{T <: Integer, f} <: Real
+    i::T
+
+    # constructor for manipulating the representation;
+    # selected by passing an extra dummy argument
+    FixedPoint(y::T, _) = new(i)
+    FixedPoint(i::Integer,_) = new(i % T)
+
+    FixedPoint(x) = convert(FixedPoint{T,f}, x)
+end
+
+# basic typealiases for signed and unsigned
+typealias Fixed{T <: Signed, f} FixedPoint{T, f}
+typealias UFixed{T <: Unsigned, f} FixedPoint{T, f}
+
+# default provided typealiases
+typealias Fixed16 Fixed{Int32, 16}
+typealias UFixed8  UFixed{UInt8,8}
+typealias UFixed10 UFixed{UInt16,10}
+typealias UFixed12 UFixed{UInt16,12}
+typealias UFixed14 UFixed{UInt16,14}
+typealias UFixed16 UFixed{UInt16,16}
+
 reinterpret(x::FixedPoint) = x.i
+reinterpret{T,f}(::Type{T}, x::FixedPoint{T,f}) = x.i
+reinterpret{T <: Integer,f}(::Type{FixedPoint{T,f}}, x::T) = FixedPoint{T,f}(x, 0)
+
+rawtype{T,f}(::Type{FixedPoint{T,f}}) = T
+nbitsfrac{T,f}(::Type{FixedPoint{T,f}}) = f
+rawone(v) = reinterpret(one(v))
 
-# comparison
-=={T <: FixedPoint}(x::T, y::T) = x.i == y.i
- <{T <: FixedPoint}(x::T, y::T) = x.i  < y.i
-<={T <: FixedPoint}(x::T, y::T) = x.i <= y.i
+# comparisons
+=={T <: FixedPoint}(x::T, y::T) = reinterpret(x) == reinterpret(y)
+ <{T <: FixedPoint}(x::T, y::T) = reinterpret(x)  < reinterpret(y)
+<={T <: FixedPoint}(x::T, y::T) = reinterpret(x) <= reinterpret(y)
 
 # predicates
 isinteger{T,f}(x::FixedPoint{T,f}) = (x.i&(1<<f-1)) == 0
+isfinite(x::FixedPoint) = true
+isnan(x::FixedPoint) = false
+isinf(x::FixedPoint) = false
 
+# traits
 typemax{T<: FixedPoint}(::Type{T}) = T(typemax(rawtype(T)), 0)
 typemin{T<: FixedPoint}(::Type{T}) = T(typemin(rawtype(T)), 0)
 realmin{T<: FixedPoint}(::Type{T}) = typemin(T)
 realmax{T<: FixedPoint}(::Type{T}) = typemax(T)
+eps{T<:FixedPoint}(::Type{T}) = T(one(rawtype(T)),0)
+eps{T<:FixedPoint}(::T) = eps(T)
+sizeof{T<:FixedPoint}(::Type{T}) = sizeof(rawtype(T))
 
-include("fixed.jl")
-include("ufixed.jl")
-include("deprecations.jl")
+zero{T <: FixedPoint}(::Type{T}) = T(zero(rawtype(T)),0)
+ one{T <: FixedPoint}(::Type{T}) = T(2^nbitsfrac(T)-1,0)
```

Comment:
I just noticed that this is probably problematic for `FixedPoint{Int32,32}`. I doubt we can do anything about that, however, without tanking performance elsewhere.